src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -339,12 +340,301 @@ void LSHSearch<SortPolicy>::BaseCase(const size_t queryIndex,
         referenceIndex, distance);
 }
 
+
+// Compare class for <double, size_t> pair, used in GetAdditionalProbingBins.
+class CompareGreater
+{
+  public:
+    bool operator()(
+        std::pair<double, size_t> p1,
+        std::pair<double, size_t> p2){
+      //only compare the double values
+      return p1.first > p2.first;
+    }
```

Comment:
You are right - I started implementing that, but then somewhere in the middle found a bug with my initial code so I never got around to throwing CompareGreater away.

Yes, in the very unlikely event where two scores are exactly the same, it doesn't really matter which one gets priority anyway - so ordering the second element is as arbitrary as anything else.